services/device-registry-base/src/main/java/org/eclipse/hono/service/management/BaseDto.java

```diff
@@ -0,0 +1,208 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.service.management;
+
+import java.time.Instant;
+import java.util.function.Supplier;
+
+import org.eclipse.hono.annotation.HonoTimestamp;
+import org.eclipse.hono.util.RegistryManagementConstants;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * The base class for implementing a DTO (Data Transfer Object) to store data in registry implementations.
+ *
+ * @param <T> The type of the data object
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
```

Comment:
Makes sense. As far as I can see the only subclass which actually requires that property would be `CredentialsDto` as its `requiresMerging` is not supposed to be persisted.

Reply:
I think the reason why this property is there (this is also only migrated from the old BaseDto class) is: all of our MongoDb collections have an "_id" property. When reading data from MongoDb `JsonObject.mapTo()` is called which then would fail without that annotation.

Hence allowing sub-classes to decide on their own would be possible, but all DTOs (except the one for the file-based implementation would require adding at least `@JsonIgnoreProperties("_id")` or we introduce a "MongoDbBaseDto" from which the other MongoDb DTOs are derived which I would do rather than duplicating this in every DTO.